fvm/src/syscalls/debug.rs

```diff
@@ -21,3 +23,36 @@ pub fn enabled(context: Context<'_, impl Kernel>) -> Result<i32> {
         -1
     })
 }
+
+// TODO: make output path more configurable, maybe add extra gaurds/limitations
+pub fn capture_artifact(
+    context: Context<'_, impl Kernel>,
+    name_off: u32,
+    name_len: u32,
+    data_off: u32,
+    data_len: u32,
+) -> Result<()> {
+    // No-op if disabled.
+    if !context.kernel.debug_enabled() {
+        return Ok(());
+    }
+    let src_dir = std::env::current_dir()
+        .or_fatal()?
+        .canonicalize()
+        .or_fatal()?
+        .join("artifacts");
+    DirBuilder::new()
+        .recursive(true)
+        .create(src_dir.clone())
+        .or_fatal()?;
+
+    let data = context.memory.try_slice(data_off, data_len)?;
+    let name = context.memory.try_slice(name_off, name_len)?;
+    let name = String::from_utf8(name.to_owned())
+        .or_error(fvm_shared::error::ErrorNumber::IllegalArgument)?;
```

Comment:
1. No need for `to_owned`.
2. We should enforce some _basic_ naming rules:
    1. Less than 256 bytes.
    2. Single path component (make sure `path::is_separator` is false for all bytes).
    3. Doesn't start with a `.`.

Reply:
iv. is a least one char